presto-main/src/main/java/com/facebook/presto/execution/buffer/OutputBufferMemoryManager.java

```diff
@@ -153,7 +158,19 @@ public long getPeakMemoryUsage()
     public synchronized void close()
     {
         updateMemoryUsage(-bufferedBytes.get());
-        systemMemoryContextSupplier.get().close();
+        getSystemMemoryContext().ifPresent(LocalMemoryContext::close);
         closed = true;
     }
+
+    private Optional<LocalMemoryContext> getSystemMemoryContext()
+    {
+        try {
+            return Optional.of(systemMemoryContextSupplier.get());
+        }
+        catch (Exception ignored) {
+            // This is possible with races, e.g., a task is created and then immediately aborted,
+            // so that the task context hasn't been created yet (as a result there's no memory context available).
```

Comment:
Does it throw or does it return null?  If the latter, use `ofNullable`.

Reply:
It throws.